### ISSUE TITLE
calculate embeddings for several proteins with the new model gLM2

### ISSUE THREAD

**Issue by marioernestovaldes** [TattaBio/gLM2#8]
Hi @scopello @sokrypton great work! I have a question.

I used your first model gLM to generate embeddings for different proteins. The generated tensors were (number of proteins x number of "hidden states) (in my case, I was performing some testing with 100 proteins, and the tensors were 100x1280)

Now with this new model gLM2, the generated tensors are (number of proteins x number of tokens x number of "hidden states) (in my case, I was performing some testing with 100 proteins and using 4096 tokens, and the tensors were 100x4096x1280).

In the first scenario, performing downstream analysis with the generated embeddings was relatively straightforward as the results were a 2D matrix. Now, with the new model, the matrix is 3D and I am not sure how to proceed. Do you have any recommendations? 

Thank in advance...

**Comment by scopello** [comment]
Hi @marioernestovaldes, thanks for the question. You can mean pool the embeddings across the `seq_len` dimension to go from `[batch, seqlen=4096, D=1280] -> [batch, D=1280]`. Example code:

```
import torch
from transformers import AutoModel, AutoTokenizer

model = AutoModel.from_pretrained('tattabio/gLM2_650M', torch_dtype=torch.bfloat16, trust_remote_code=True).cuda()
tokenizer = AutoTokenizer.from_pretrained('tattabio/gLM2_650M', trust_remote_code=True)

sequences = ["<+>MALTKVEKRNRIKRRVRGKISGTQASPRLSVYKSNK<+>aatttaaggaa<->MLGIDNIERVKPGGLELVDRLVAVNRVTKVTKGGRAFGFSAIVVVGNED",
             "<+>MALTKVEKRNRIKRRVRGKISGT",
             "<->MLGIDNIERVKPGGLELVDRLVAVNRVTKVTKGGRAFGFSAIVVVGNED",]

encodings = tokenizer(sequences, return_tensors='pt', padding=True)

# Extract batch embeddings.
with torch.no_grad():
    attention_mask = encodings.attention_mask.bool().cuda()
    embeddings = model(encodings.input_ids.cuda(), attention_mask=attention_mask, output_hidden_states=True).last_hidden_state

# Mean pool. Shape will go from [batch, seqlen, D] -> [batch, D]
mask = attention_mask.unsqueeze(-1)
pooled_embeds = torch.where(mask, embeddings, 0.0)
pooled_embeds = torch.sum(pooled_embeds, 1) / torch.sum(mask, dim=1, dtype=pooled_embeds.dtype)
